Framework/Nexus/inc/MantidNexus/NexusClasses.h

```diff
@@ -866,6 +868,10 @@ class DLLExport NXData : public NXMainClass {
   NXFloat openFloatData() { return openData<float>(); }
   /// Opens data of int type
   NXInt openIntData() { return openData<int>(); }
+  /// Opens data of size type
+  NXSize openSizeData() { return openData<std::size_t>(); }
+  /// Opens data of unsigned int type
+  NXUInt openUIntData() { return openData<unsigned int>(); }
```

Comment:
The entries in the nexus files are specifically unsigned int (and not unsigned long) type, so when I try to read with openSizeData I get spurious numbers wrongly read from memory.